### PR TITLE
[#9025] fix keycard pairings persistence

### LINKED ISSUE #9025: Can't proceed with creating account if cancel process on "Confirm seed phrase"
###  Description

[comment]: # (Feature or Bug? i.e Type: Bug)
*Type*: Bug
*Summary*: can't pair the card if cancel process on "Backup seed phrase" stage - nothing happens if enter pairing code.

#### Expected behavior
can proceed 

#### Actual behavior
nothing happens

### Reproduction

- Open Status
- Start creating account with keycard
- On step when should confirm word from seed phrase, tap "Cancel"

### Additional Information

[comment]: # (Please do your best to fill this out.)
- Status version: nightly 
- Operating System: Android

#### Logs

Logcat: [no_pairing.log](https://github.com/status-im/status-react/files/3642471/no_pairing.log)

Keycard info:
```
INFO [09-23|16:05:14.339] waiting for a card                       package=status-go/cmd/keycard
INFO [09-23|16:05:14.342] card found                               package=status-go/cmd/keycard index=0
DEBUG[09-23|16:05:14.342] using reader                             package=status-go/cmd/keycard name="ACS ACR 38U-CCID"
DEBUG[09-23|16:05:14.342] connecting to card                       package=status-go/cmd/keycard reader="ACS ACR 38U-CCID"
DEBUG[09-23|16:05:14.344] card protocol                            package=status-go/cmd/keycard T=1
INFO [09-23|16:05:14.344] info started                             package=status-go/cmd/keycard
INFO [09-23|16:05:14.344] select keycard applet                    package=status-go/cmd/keycard
DEBUG[09-23|16:05:14.344] apdu command                             package=io                    hex="00 A4 04 00 09 A0 00 00 08 04 00 01 01 01 00"
DEBUG[09-23|16:05:14.375] apdu response                            package=io                    hex="A4 61 8F 10 33 F0 C0 8B CA 61 32 73 31 6F 9F C7 ED 49 32 6F 80 41 04 DC EC C9 4A 8B 97 69 7E 93 9E 61 74 57 C3 89 4D 65 E5 A8 A0 37 DF 84 A5 1C B4 6B FA 2D 96 72 24 4E 89 8E 92 10 37 80 DC 92 42 FA 96 5C 91 BD 42 51 E4 01 36 0F BD B3 38 D4 87 BA C4 FC 24 79 95 02 02 02 02 02 01 04 8E 00 8D 01 0F 90 00"
Installed: true
Initialized: true
InstanceUID: 0x33f0c08bca613273316f9fc7ed49326f
SecureChannelPublicKey: 0x04dcecc94a8b97697e939e617457c3894d65e5a8a037df84a51cb46bfa2d9672244e898e92103780dc9242fa965c91bd4251e401360fbdb338d487bac4fc247995
Version: 0x0202
AvailableSlots: 0x04
KeyUID: 0x
Capabilities:
  Secure channel:true
  Key management:true
  Credentials Management:true
  NDEF:true
```

### PR DESCRIPTION
**Pull request by dmitryn**
fixes #9025

### Summary

* Update pairings data in app-db after successful pair
* Fix pairings retrieval

status: ready

### CODE REVIEW

File: src/status_im/hardwallet/core.cljs [hunk 1]
```diff
@@ -1505,15 +1505,16 @@
         instance-uid (get-in db [:hardwallet :application-info :instance-uid])
```
Comment (yenda) [inline src/status_im/hardwallet/core.cljs:1511]: could you add a docstring to on-pair-success that explains what a pairing is and why we need to persist it here?

Comment (dmitryn) [reply src/status_im/hardwallet/core.cljs:1511]: added

File: src/status_im/hardwallet/core.cljs [hunk 2]
```diff
@@ -1499,21 +1499,27 @@
   {:db (assoc-in db [:hardwallet :pairings] pairings)})
 
 (fx/defn on-pair-success
+  "When pairing to device completed, we need to persist pairing data to
```
Comment (yenda) [inline src/status_im/hardwallet/core.cljs]:
```suggestion
  "When pairing to device has completed, we need to persist pairing data to
```

File: src/status_im/hardwallet/core.cljs [hunk 3]
```diff
@@ -1499,21 +1499,27 @@
...
+  local storage. That's needed for to ensure that during keycard setup
```
Comment (yenda) [inline src/status_im/hardwallet/core.cljs]:
```suggestion
  local storage. That's needed to ensure that during keycard setup
```

File: src/status_im/hardwallet/core.cljs [hunk 4]
```diff
@@ -1499,21 +1499,27 @@
...
+  keycard won't run out of pairings slots, ie. we don't pair same
```
Comment (yenda) [inline src/status_im/hardwallet/core.cljs]:
```suggestion
  keycard won't run out of pairings slots, ie. we don't pair the same
```

File: src/status_im/hardwallet/core.cljs [hunk 5]
```diff
@@ -1499,21 +1499,27 @@
...
+  card to same device more than one time. Also, this allows user to proceed
```
Comment (yenda) [inline src/status_im/hardwallet/core.cljs]:
```suggestion
  card to the same device more than one time. Also, this allows the user to proceed
```

File: src/status_im/hardwallet/core.cljs [hunk 6]
```diff
@@ -1499,21 +1499,27 @@
...
+  with setup skiping pairing step if pairing was already done during previous
```
Comment (yenda) [inline src/status_im/hardwallet/core.cljs]:
```suggestion
  with setup and skip the pairing step if the pairing was already done during a previous
```

**Comment by yenda** [comment]
lgtm squash and rebase before we test it please

**Review by flexsurfer — APPROVED** [review]

**Pull request merged**